src/java/org/apache/cassandra/config/RetrySpec.java

```diff
@@ -0,0 +1,182 @@
+    {
+        public static final MaxAttempt DISABLED = new MaxAttempt();
+
+        public int value;
+
+        public MaxAttempt(int value)
+        {
+            if (value < 1)
+                throw new IllegalArgumentException("max attempt must be positive; but given " + value);
+            this.value = value;
+        }
+
+        private MaxAttempt()
+        {
+            value = 0;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) return true;
+            if (o == null) return false;
+            if (o instanceof Integer) return this.value == ((Integer) o).intValue();
+            if (getClass() != o.getClass()) return false;
+            MaxAttempt that = (MaxAttempt) o;
+            return value == that.value;
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(value);
+        }
+
+        @Override
+        public String toString()
+        {
+            return Integer.toString(value);
+        }
+    }
+
+    public static class Partial extends RetrySpec
+    {
+        public Partial()
+        {
+            this.type = null;
+            this.maxAttempts = null;
+            this.baseSleepTime = null;
+            this.maxSleepTime = null;
+        }
+
+        public RetrySpec withDefaults(RetrySpec defaultValues)
+        {
+            Type type = nonNull(this.type, defaultValues.getType(), DEFAULT_TYPE);
+            MaxAttempt maxAttempts = nonNull(this.maxAttempts, defaultValues.getMaxAttempts(), DEFAULT_MAX_ATTEMPTS);
+            LongNanosecondsBound baseSleepTime = nonNull(this.baseSleepTime, defaultValues.getBaseSleepTime(), DEFAULT_BASE_SLEEP);
+            LongNanosecondsBound maxSleepTime = nonNull(this.maxSleepTime, defaultValues.getMaxSleepTime(), DEFAULT_MAX_SLEEP);
+            return new RetrySpec(type, maxAttempts, baseSleepTime, maxSleepTime);
+        }
+
+        private static <T> T nonNull(@Nullable T left, @Nullable T right, T defaultValue)
+        {
+            if (left != null)
+                return left;
+            if (right != null)
+                return right;
+            return defaultValue;
+        }
+    }
+
+    private static final Type DEFAULT_TYPE = Type.Expoential;
+    private static final MaxAttempt DEFAULT_MAX_ATTEMPTS = MaxAttempt.DISABLED;
+    private static final LongNanosecondsBound DEFAULT_BASE_SLEEP = new LongNanosecondsBound("200ms");
+    private static final LongNanosecondsBound DEFAULT_MAX_SLEEP = new LongNanosecondsBound("1s");
+
+    public Type type = DEFAULT_TYPE;
+    /**
+     * Represents how many retry attempts are allowed.  If the value is 2, this will cause 2 retries + 1 original request, for a total of 3 requests!
+     * <p/>
+     * To disable, set to 0.
+     * <p/>
+     * To make it unbound, used -1
+     */
+    public MaxAttempt maxAttempts = DEFAULT_MAX_ATTEMPTS; // 2 retries, 1 original request; so 3 total
```

Comment:
```suggestion
    public MaxAttempt max_attempts = DEFAULT_MAX_ATTEMPTS; // 2 retries, 1 original request; so 3 total
```
I guess we might want to use underscores if these names are going to show up literally in the YAML?

Reply:
the yaml layer makes `maxAttempts` expose as `max_attempts`; so this isn't needed... I added that something like 2 years ago, but think people just keep using their habits

Reply:
have tests for these new configs to show that this is true ^_^